govcd/nsxt_edgegateway_static_route.go

```diff
@@ -0,0 +1,269 @@
+	// edgeGatewayId is stored for usage in NsxtEdgeGatewayStaticRoute receiver functions
+	edgeGatewayId string
+}
+
+// CreateStaticRoute based on type definition
+func (egw *NsxtEdgeGateway) CreateStaticRoute(staticRouteConfig *types.NsxtEdgeGatewayStaticRoute) (*NsxtEdgeGatewayStaticRoute, error) {
+	client := egw.client
+	endpoint := types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointEdgeGatewayStaticRoutes
+	apiVersion, err := client.getOpenApiHighestElevatedVersion(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	// Insert Edge Gateway ID into endpoint path
+	urlRef, err := client.OpenApiBuildEndpoint(fmt.Sprintf(endpoint, egw.EdgeGateway.ID))
+	if err != nil {
+		return nil, err
+	}
+
+	returnObject := &NsxtEdgeGatewayStaticRoute{
+		client:                     egw.client,
+		edgeGatewayId:              egw.EdgeGateway.ID,
+		NsxtEdgeGatewayStaticRoute: &types.NsxtEdgeGatewayStaticRoute{},
+	}
+
+	// Non standard behavior of entity - `Details` field in task contains ID of newly created object while the Owner is Edge Gateway
+	task, err := client.OpenApiPostItemAsync(apiVersion, urlRef, nil, staticRouteConfig)
+	if err != nil {
+		return nil, fmt.Errorf("error creating NSX-T Edge Gateway Static Route: %s", err)
+	}
+
+	err = task.WaitTaskCompletion()
+	if err != nil {
+		return nil, fmt.Errorf("error creating NSX-T Edge Gateway Static Route: %s", err)
+	}
+
+	// API does not return an ID for created object - we know that it is expected to be in
+	// task.Task.Details therefore attempt to find it, but if it is empty - look for an entity by a
+	// set of requested parameters
+	staticRouteId := task.Task.Details
+	if staticRouteId != "" {
+		getUrlRef, err := client.OpenApiBuildEndpoint(fmt.Sprintf(endpoint, egw.EdgeGateway.ID), staticRouteId)
+		if err != nil {
+			return nil, err
+		}
+		err = client.OpenApiGetItem(apiVersion, getUrlRef, nil, returnObject.NsxtEdgeGatewayStaticRoute, nil)
+		if err != nil {
+			return nil, fmt.Errorf("error retrieving NSX-T Edge Gateway Static Route after creation: %s", err)
+		}
+	} else {
+		// ID was not present in response, therefore Static Route needs to be found manually. Using
+		// 'Name', 'Description' and 'NetworkCidr' for finding the entity. Duplicate entries can
+		// exist, but but it should be a good enough combination for finding unique entry until VCD API is fixed
+		allStaticRoutes, err := egw.GetAllStaticRoutes(nil)
+		if err != nil {
+			return nil, fmt.Errorf("error retrieving NSX-T Edge Gateway Static Route after creation: %s", err)
+		}
+
+		var foundStaticRoute bool
+		for _, singleStaticRoute := range allStaticRoutes {
+			if singleStaticRoute.NsxtEdgeGatewayStaticRoute.Name == staticRouteConfig.Name &&
+				singleStaticRoute.NsxtEdgeGatewayStaticRoute.NetworkCidr == staticRouteConfig.NetworkCidr &&
+				singleStaticRoute.NsxtEdgeGatewayStaticRoute.Description == staticRouteConfig.Description {
+				foundStaticRoute = true
+				returnObject = singleStaticRoute
```

Comment:
Should it `break` here? (as you already found a candidate)

Reply:
Fixed and ran sdk and terraform tests